### ISSUE TITLE
[Feature request] The Notepad++ Curveball: When Commands Don’t Follow Your Lead

### ISSUE THREAD

**Issue by Bartolomeus-649** [notepad-plus-plus/notepad-plus-plus#15275]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

Sometimes Notepad++ ignores your selected text when you run a command, which can be pretty frustrating.

### Describe the solution you'd like.

The behavior of certain commands in Notepad++ is currently limited to the active line, which can be counterintuitive when users have selected a block of text. This feature request is to change the default behavior so that all commands would apply to the user's selected text.

Proposed Change:
Modify the default behavior of commands to affect the entire block of selected text. This would make the user experience more intuitive and efficient, as commands would act on all highlighted text, not just the active line.

Optional Setting:
For users who rely on the existing behavior, an optional setting titled "Apply commands to selection if possible" could be introduced. This setting would allow those users to revert to the original command behavior if they choose.

Rationale:
Users often expect that when they select text and execute a command, the action will apply to their entire selection. The proposed change would align Notepad++ with these expectations, thereby improving usability and reducing confusion.

Impact:
This enhancement would streamline the editing process in Notepad++, making it more user-friendly and efficient.

### Debug Information

```shell
N/A
```

**Comment by alankilborn** [comment]
>Modify the default behavior of commands to affect the entire block of selected text.

I don't think this can be taken seriously as a "blanket" thing, i.e., "...behavior of ALL commands should be...".

If you want to specifically list the commands whose current actions bother you, that would probably be better.

**Comment by Bartolomeus-649** [comment]
which commands that modify or apply an effect to text would you not want to be applied to only the selected text instead of to just the current line or the entire text?

Isn't it likely that if a user bothered to select some text before issuing a command, then that user want to apply that command to that selection?

Many commands use the current selection and only modify or apply an effect to the text within it.
I argue that the same reason those commands only modify or apply an effect to the selected text should be used for all other commands which ignore the selection.

An example of a command which ignore the current selection is **Collapse/Uncollapse current level** on the View-menu.

**Comment by alankilborn** [comment]
@Bartolomeus-649 

Now that I ponder it further... IIRC, someone, and I think it was @ArkadiuszMichalski , spent a good deal of effort recently (in the past year or so) working on what you're asking for.  I believe _some_ of his efforts were accepted into Notepad++, but definitely not in a blanket way; it turned out it is more of a difficult "problem" to solve than simply saying "Modify the default behavior of commands to affect the entire block of selected text. "

**Comment by Bartolomeus-649** [comment]
I can't see why it should not be rather straight forward. Either the effect is for the current line, in which case you could just do a `foreach(row in selection) ApplyCommand(row)`. 
Or it applies to the entire text, in which case the selected text could be put into a temporary new document, and then the command could be applied to that entire text there, then the old text is deleted, and the modified text inserted in its place.

Sure, there will be some cases where it is tricky, but don't let those block the entire thing, do the easy first.

**Comment by alankilborn** [comment]
>I can't see why it should not be rather straight forward

Users often make this mistake in thinking.

---

As I said, someone has previously spent a lot of time/effort on it, with only partial success.
If you'd like to try your hand at it, due to what you think is its straightforward nature, feel free -- Notepad++ accepts user contributed code (but there's no guarantee of acceptance).

**Comment by alankilborn** [comment]
@Bartolomeus-649 

Another thing I just noticed:  Your issue title isn't likely to get developers interested in your issue, as it appears to be sarcasm/criticism.  You might want to rename it (titles are editable by their creator) to something more indicative of what you issue actually is.

**Comment by myTempTemp** [comment]
侯今吾是一条狗 我家里有一条狗，我给它起名叫侯今吾，来吃点屎吧！
